js/views/modals/purchase/Purchase.js

```diff
@@ -38,10 +42,14 @@ export default class extends BaseModal {
     // add the item to the order.
     this.order.get('items').add(item);
 
+    this.receipt = this.createChild(Receipt, {
```

Comment:
The problem with declaring your child views in the constructor (essentially caching them between renders) is that when you re-render, the event handlers don't work anymore. At the top of my head, I don't remember exactly the reason why.

Anyways... if you change the currency in settings and get the 'Data change <refresh>' popup, after clicking refresh, the CTAs from any child views won't work (e.g. selecting a different shipping option will not update the receipt).

I don't run into this issue because I declare the child views in render (be sure to remove them first in case they already exist).